crates/subspace-service/src/lib.rs

```diff
@@ -559,11 +555,11 @@ where
 
     let segment_header_cache = SegmentHeaderCache::new(client.clone());
 
-    let (node, bootstrap_nodes) = match config.subspace_networking.clone() {
+    let (node, bootstrap_nodes, piece_cache) = match config.subspace_networking.clone() {
         SubspaceNetworking::Reuse {
             node,
             bootstrap_nodes,
-        } => (node, bootstrap_nodes),
+        } => (node, bootstrap_nodes, None),
```

Comment:
I don't think this is correct. If piece cache size is a function of the plot size, then even if networking is provided from the outside, piece cache should still work and be functional, it'll just be potentially composed of node + farmer cache sizes (but they will cache different pieces due to different policies).

Let's have at least a TODO here because this is problematic, it seems to make SDK and its users incredibly useless for caching purposes on the network.

Reply:
Not sure what you mean about SDK requirements, this is a core protocol requirement: regardless of how you run a farmer, it should provide piece cache, but now it will not in case of SDK unless I misunderstand something.